networkit/cpp/io/test/MemoryMappedFileGTest.cpp

```diff
@@ -0,0 +1,100 @@
+/*
+* MemoryMappedFileGTest.h
+*
+*  Created on: 28.07.2018
+*      Author: Manuel Penschuck (networkit@manuel.jetzt)
+*/
+
+
+#include <algorithm>
+#include <fstream>
+#include <limits>
+#include <memory>
+#include <random>
+#include <string>
+#include <stdexcept>
+
+#include "MemoryMappedFileGTest.h"
+#include "../MemoryMappedFile.h"
+
+namespace NetworKit {
+	// This object generates a temporary file and fills it with a request number
+	// of random bytes. On destruction of the elemen, the file is deleted automatically.
+	// Make sure that no application holds a handle to the file at this point, since
+	// on Windows removal of the file fails silently in this case.
+	class TemporaryFile {
+	public:
+		using value_type = char;
+		using const_iterator = const value_type*;
+
+		TemporaryFile(size_t bytes, std::default_random_engine& prng) : 
+			dataBegin(new value_type[bytes]),
+			dataEnd(dataBegin.get() + bytes)
+		{
+			// obtain temporary file name (C++17 offers std::filesystem, which we currently cannot rely on)
+			while (true) {
+				std::uniform_int_distribution<unsigned> distr;
+				path = "tempfile_" + std::to_string(distr(prng));
+
+				std::ifstream f(path);
+				if (!f.good()) break;
```

Comment:
There is a TOCTTOU condition here (the file might be created by a different program after the check succeeds).